src/Mesh/Topologies.jl

```diff
@@ -1010,48 +1010,48 @@ end
 
 
 """    
-    grid_stretching_1d(coord_min, coord_max, Ne, stretching_type)
+    grid1d(a, b[, stretch::AbstractGridStretching]; elemsize, nelem)
 
-        This function is the extrema of a 1D domain (e.g. the x3 direction of
-        the mesh at hand) and stretches the 1D grid in that direction.
+Discretize the 1D interval [`a`,`b`] into elements.
+Exactly one of the following keyword arguments must be provided:
+- `elemsize`: the average element size, or
+- `nelem`: the number of elements.
 
-        It returns the 1D range `range_stretched` to be then passed to
-        `brickrange = (x1_range, x2_range, x3_range)` in the driver in place of
-        `x1_range`, or `x2_range` or `x3_range`
+The optional `stretch` argument allows stretching, otherwise the element sizes will be uniform.
 
-        The use needs to define the type of stretching `stretching_type` 
-        Now `boundary_layer` and `top_layer` are the only options availabe.
-
-        Add more functions to the function Mesh.Topologies.grid_stretching_1d.
-        
+Returns either a range object or a vector containing the element boundaries.
 """
+function grid1d(a, b, stretch=nothing; elemsize=nothing, nelem=nothing)
+  xor(nelemN === nothing, elemsize === nothing) || error("Either `elemsize` or `nelem` arguments must be provided")
+  if elemsize !== nothing
+    nelem = round(Int,abs(b-a)/elemsize)
+  end
+  grid1d(a, b, stretch, nelem)
+end
+function grid1d(a, b, ::Nothing, nelem)
+  range(a, stop=b, length=N+1)
+end
 
-function grid_stretching_1d(coord_min, coord_max, Ne, stretching_type, attractor_value=0)
-
-    DFloat = eltype(coord_min)
-    
-    #build physical range to be stratched
-    range_stretched = range(DFloat(coord_min), length = Ne + 1, DFloat(coord_max))
-   
-    #build logical space
-    s  = range(DFloat(0), length=Ne[1]+1, DFloat(1))
-
-    stretch_coe = 0.0
-    if (stretching_type == "boundary_stretching")
-        stretch_coe = 2.5
-        range_stretched = (coord_max - coord_min).*(exp.(stretch_coe * s) .- 1.0)./(exp(stretch_coe) - 1.0)
-    elseif (stretching_type == "top_stretching")
-        stretch_coe = 2.5
-        range_stretched = -(coord_max - coord_min).*(exp.(stretch_coe * s) .- 1.0)./(exp(stretch_coe) - 1.0)
```

Comment:
@smarras79 I couldn't quite figure out what this was doing. Do you have some references for these?

Reply:
@simonbyrne these functions simply move the nodes towards the top or bottom boundaries of the domain.
They are generally referred to as  "grid stretching functions" in the grid generation literature.

Reply:
Is there a decent reference I can use? I would like to document these.

Reply:
"Handbook of Grid Generation"  J. F. Thompson,  B. K. Soni, N. P. Weatherill (Editors) RCR Press 1999

Reply:
Do you know where in the book? I wasn't able to find them.

Reply:
I don't have a copy of the book anymore, but if tyou can't find them there, you will certainly find them here:

`Computational Fluid Mechanics And Heat` Pletcher, Tennehill, Anderson`. They are juist standard exponential stretching function used in CFD regardless of the numerical method.

Reply:
ah, nevermind.

Reply:
I couldn't find a reference for the internal one. If you know of anything, let me know.

Reply:
You don't need a reference for these functions. Anyone can come up with any arbitrary function with some specific properties to move the nodes where needed. I wouldn't worry about it at all.

Reply:
Thanks for merging it all together to the new code, btw!